src/basic/pidref.c

```diff
@@ -143,3 +144,30 @@ int pidref_kill_and_sigcont(PidRef *pidref, int sig) {
 
         return 0;
 }
+
+int pidref_sigqueue(PidRef *pidref, int sig, int value) {
+
+        if (!pidref)
```

Comment:
Not a matter, but `pidref_is_set()` is better to merge with the return at the end of this function.

Reply:
not sure I follow. this mimics behaviour in pidref_kill().

Reply:
I mean,
```c
int pidref_sigqueue(PidRef *pidref, int sig, int value) {
        if (!pidref_is_set(pidref))
                return -ESRCH;

        if (pidref->fd > 0) {
                ...;
        }

        assert(pidref->pid > 0);
        return RET_ERRNO(sigque(...));
}
```

Not necessary to apply my suggestion, yes as it follows `pidref_kill()`. But, I usually want to write functions first exclude unexpected situations, then do requested operations. Feel free to ignore the comment.